### PR TITLE
Adds few destroy functions

### PR DESCRIPTION
**Pull request by bishtgautam**
To support AMR, few destroy functions are added that can be called when the mesh is refined.

### CODE REVIEW

File: include/private/rdydmimpl.h [hunk 1]
```diff
@@ -19,5 +19,6 @@ PETSC_INTERN PetscErrorCode CloneAndCreateCellCenteredDM(DM dm, const SectionFie
 PETSC_INTERN PetscErrorCode CreateDM(RDy rdy);
 PETSC_INTERN PetscErrorCode CreateAuxiliaryDM(RDy rdy);
 PETSC_INTERN PetscErrorCode CreateVectors(RDy rdy);
+PETSC_INTERN PetscErrorCode CreateSectionForSWE(RDy, PetscSection *);
```
Comment (jeff-cohere) [inline include/private/rdydmimpl.h:22]: In the interest of confining SWE-specific code to `rdysweimpl.h` and `src/swe`, maybe we could call this `CreateSection` and have it dispatch to `CreateSWESection` in `rdysweimpl.h` (defined in `src/swe/physics_swe.c`, perhaps).

File: src/rdycore.c [hunk 2]
```diff
@@ -117,6 +117,63 @@ PetscErrorCode RDyCreateF90(MPI_Fint *f90_comm, const char *config_file, RDy *rd
...
+    if ((*rdy)->regions[i].cell_ids) {
+      PetscFree((*rdy)->regions[i].cell_ids);
+    }
+  }
+  if ((*rdy)->regions) PetscFree((*rdy)->regions);
+
+  PetscFunctionReturn(PETSC_SUCCESS);
+}
+
+/// @brief Destroy boundariy data structures
```
Comment (jeff-cohere) [inline src/rdycore.c:150]: boundariy -> boundary

File: include/private/rdydmimpl.h (hunk 1 above)
Comment (bishtgautam) [reply include/private/rdydmimpl.h:22]: I agree with the change you have suggested, but how about you make that change in the next operator refactor PR?

**Review by jeff-cohere — APPROVED** [review]
Looks good to me. I'll tuck the SWE stuff out of sight in the next operator refactor PR.

**Pull request merged**